test/externalTests/common.sh

```diff
@@ -153,27 +153,25 @@ function force_solc
 
     printLog "Forcing solc version..."
     cat >> "$config_file" <<EOF
-module.exports['compilers'] = {solc: {version: "$dir/solc"} };
+module.exports['compilers'] = {solc: {version: "$dir/solc", settings: $(solc_settings "$OPTIMIZER_LEVEL" istanbul)}};
```

Comment:
This is a bit confusing at least for me because function is called `force_solc` but it is also forcing a setting. There is another function called `force_solc_settings` that is also doing the same?

Reply:
Yeah, that part is confusing. I didn't want to refactor too much in this PR because it's mostly meant as fix but since you pointed it out, I added a few more commits that clean this up.

I have renamed these functions and removed `force_solc` so that now you need to separately change the config and overwrite `solc` npm module. The only functional change is the fact that now there's no special initial config - instead it always specifies all the options (i.e. both the compiler path/version and the optimizer settings). I put that one in a commit separate from others that are pure refactors.